### PR TITLE
Updated Course Migrator and Publisher

### PR DESCRIPTION
**Pull request by clintonb**
- Migrator now reads and stores the course mode expiration datetime
- Publisher now publishes the course mode expiration datetime

XCOM-478

@jimabramson @rlucioni @Nickersoft 
Review #195 first

### CODE REVIEW

File: ecommerce/courses/publishers.py [hunk 1]
```diff
@@ -15,7 +15,8 @@ def serialize_seat_for_commerce_api(self, seat):
             'name': seat.attr.certificate_type,
             'currency': stock_record.price_currency,
             'price': int(stock_record.price_excl_tax),
-            'sku': stock_record.partner_sku
+            'sku': stock_record.partner_sku,
+            'expires': seat.expires.isoformat() if seat.expires else None,
```
Comment (jimabramson) [inline ecommerce/courses/publishers.py:19]: is that going to come out as UTC, or whatever django decides the requestor's timezone is ?

Comment (clintonb) [inline ecommerce/courses/publishers.py:19]: It should be UTC; however, on the slim chance it is not, `isoformat()` includes the timezone offset (e.g. +00:00).

**Comment by rlucioni** [comment]
:+1:

File: ecommerce/courses/publishers.py (hunk 1 above)
Comment (jimabramson) [inline ecommerce/courses/publishers.py:19]: don't mean to be a pain, but isoformat doesn't include the offset unless the datetime is timezone-aware, is that guaranteed?

Comment (clintonb) [inline ecommerce/courses/publishers.py:19]: Pain leads to good code...I hope. Yes, our dates are timezone-aware.

**Comment by jimabramson** [comment]
:+1:

File: ecommerce/courses/tests/test_models.py [hunk 2]
```diff
@@ -104,3 +104,12 @@ def test_save_with_publish_failure(self):
         # Reload the course from the database
         course = Course.objects.get(id=course.id)
         self.assertEqual(course.name, orignal_name)
+
+    def test_save_without_publish(self):
+        """ Verify the Course is not published to LMS is the publish kwarg is set to False. """
```
Comment (clintonb) [inline ecommerce/courses/tests/test_models.py]: Fixing a typo here.

**Pull request merged**